presto-bytecode/src/main/java/com/facebook/presto/bytecode/expression/GetFieldBytecodeExpression.java

```diff
@@ -0,0 +1,120 @@
+        extends BytecodeExpression
+{
+    private final BytecodeExpression instance;
+    private final ParameterizedType declaringClass;
+    private final String name;
+
+    public GetFieldBytecodeExpression(@Nullable BytecodeExpression instance, Class<?> declaringClass, String name)
+    {
+        this(instance, getDeclaredField(declaringClass, name));
+    }
+
+    public GetFieldBytecodeExpression(@Nullable BytecodeExpression instance, Field field)
+    {
+        this(instance, type(requireNonNull(field, "field is null").getDeclaringClass()), field.getName(), type(field.getType()));
+
+        boolean isStatic = Modifier.isStatic(field.getModifiers());
+        if (instance == null) {
+            checkArgument(isStatic, "Field is not static: %s", field);
+        }
+        else {
+            checkArgument(!isStatic, "Field is static: %s", field);
+        }
+    }
+
+    public GetFieldBytecodeExpression(@Nullable BytecodeExpression instance, FieldDefinition field)
+    {
+        this(instance, requireNonNull(field, "field is null").getDeclaringClass().getType(), field.getName(), field.getType());
+        if (instance == null) {
+            checkArgument(field.getAccess().contains(STATIC), "Field is not static: %s", field);
+        }
+        else {
+            checkArgument(!field.getAccess().contains(STATIC), "Field is static: %s", field);
+        }
+    }
+
+    public GetFieldBytecodeExpression(@Nullable BytecodeExpression instance, ParameterizedType declaringClass, String name, ParameterizedType type)
+    {
+        super(type);
+        checkArgument(instance == null || !instance.getType().isPrimitive(), "Type %s does not have fields", getType());
+        this.instance = instance;
+        this.declaringClass = requireNonNull(declaringClass, "declaringClass is null");
+        this.name = requireNonNull(name, "name is null");
+    }
+
+    @Override
+    public BytecodeNode getBytecode(MethodGenerationContext generationContext)
+    {
+        if (instance == null) {
+            return getStaticInstruction(declaringClass, name, getType());
+        }
+
+        return new BytecodeBlock()
+                .append(instance.getBytecode(generationContext))
+                .getField(declaringClass, name, getType());
+    }
+
+    @Override
+    protected String formatOneLine()
+    {
+        if (instance == null) {
+            return declaringClass.getSimpleName() + "." + name;
+        }
+        return instance + "." + name;
+    }
+
+    @Override
+    public List<BytecodeNode> getChildNodes()
+    {
+        return (instance == null) ? ImmutableList.of() : ImmutableList.of(instance);
+    }
+
+    private static Field getDeclaredField(Class<?> declaringClass, String name)
+    {
+        requireNonNull(declaringClass, "declaringClass is null");
+        requireNonNull(name, "name is null");
+
+        try {
+            return declaringClass.getField(name);
+        }
+        catch (NoSuchFieldException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)